test/helpers/codec/encoding.ts

```diff
@@ -121,7 +121,7 @@ export const signNativeTransaction = async (
   let [v, r, s] = getSignedComponents(transactionSignature).map((component) => {
     return remove0x(component)
   })
-  v = '0' + (parseInt(v, 16) - 420 * 2 - 8 - 27)
+  v = '0' + (parseInt(v, 16) - transaction.chainId * 2 - 8 - 27)
```

Comment:
I don't think that this will work for all possible chainIds, if the right hand side of the string concat results in an even length string, then v will be odd length